xypattern/pattern.py

```diff
@@ -439,22 +439,37 @@ def from_dict(json_dict: dict) -> Pattern:
 
         return pattern
 
-    def delete_range(self, x_range): 
+    def delete_range(self, x_range: tuple) -> Pattern:
         """
         Deletes data points from the pattern within the given range.
 
         :param x_range: Tuple of two floats of x values, 
-         The data points closest to these two x values remain present
-         in the pattern, but the data points in between them are deleted 
+         The data points within the tuple is deleted 
          from the pattern.
+        :return: New pattern
         """
         x, y = self.data
-        ind_min = np.argmin(np.abs(x - x_range[0]))
-        ind_max = np.argmin(np.abs(x - x_range[1]))
-        ind = np.where((x <= x[ind_min]) | (x >= x[ind_max]))
-        self._pattern_x = x[ind]
-        self._pattern_y = y[ind]
-        self.changed.emit()
+        ind = np.where((x < x_range[0]) | (x > x_range[1]))
```

Comment:
That means in the current implementation, points which lie on a limit, will be deleted. This is different than you wanted it in the morning :)

Reply:
In the old implementation, the data point closest to a provided x value remained in the pattern. It did not matter whether the provided x value was smaller or larger than the closest data point. This does matter now in the new implementation.